### ISSUE TITLE
Proposal: change BlockCreation trait and addBlock method access modifier to public

### ISSUE THREAD

**Issue by genrym** [etorreborre/specs2#685]
This will allow to extend the examples DSL to have more custom looking mutable specifications.

An example:

```
"myMethod" should {

   "affect myAnotherMethod" to {

      "perform some thingy" in {
           ...
      }

      "throw an exception" in {
         ...
      }
   }
}
```

Of course this can also be achieved by simply inlining like the following:

```
"myMethod" should {

   "affect myAnotherMethod to perform some thingy" in {
      ...
   }
   
   "affect myAnotherMethod to throw an exception" in {
         ...
   }
}
```

But then, why the DSL allows more than one level nesting? :)

**Comment by etorreborre** [comment]
So if I understand correctly you want to implement a new method `to` using `addBlock`. What is your expected output? Also you can experiment right now by using the content of `addBlock`. You can use `addFragment` and `fragmentFactory.text("xxx")`, `fragmentFactory.start`, and so on. I am not opposed to opening up `addBlock` I just want to make sure I understand your use case.

**Comment by genrym** [comment]
That's right.
I want new keywords to have exactly the same behaviour as the built in ones (`should`, `can`) but also be future proof. Meaning, if there will be changes in `should` / `can` implementations in the future, I assume they will most probably be in the `BlockCreation.addBlock` method.

Wdyt?

**Comment by etorreborre** [comment]
I think this [commit on master](https://github.com/etorreborre/specs2/commit/2fdbc7664d918e48b5f0d6b25ad576d09ff303bb) will allow you to do what you want. Please read the doc and tell me if it looks ok to you.

**Comment by genrym** [comment]
This looks great! Thanks a lot :)

**Comment by etorreborre** [comment]
It is now released as `4.3.1`.
